pkg/apiclient/apiclient.go

```diff
@@ -68,10 +68,19 @@ func NewClientFromOpts(opts Opts) (context.Context, Client, error) {
 	if opts.ArgoServerOpts.URL != "" && opts.InstanceID != "" {
 		return nil, nil, fmt.Errorf("cannot use instance ID with Argo Server")
 	}
+	var auth string
 	if opts.ArgoServerOpts.HTTP1 {
-		return newHTTP1Client(opts.ArgoServerOpts.GetURL(), opts.AuthSupplier(), opts.ArgoServerOpts.InsecureSkipVerify, opts.ArgoServerOpts.Headers)
+		if opts.AuthSupplier != nil {
```

Comment:
Should we check auth supplier and return error instead? Otherwise it's `auth=""`

Reply:
> Should we check auth supplier and return error instead? Otherwise it's `auth=""`

Of course, but we can only do something like below when there is no any Authentication.
```go
apiclient.Opts{
			ArgoServerOpts: ArgoServerOpts,
			AuthSupplier: func() string {
				return ""
			},
		})
```